### PR TITLE
Codespaces CLI release v0.1.0

### PR DESCRIPTION
**Pull request by derekbekoe**
Initial release of the Azure CLI codespaces extension.

More information on our service:
Visual Studio Codespaces - Cloud-powered development environments accessible from anywhere
https://azure.microsoft.com/en-us/services/visual-studio-online/

Closes https://github.com/microsoft/vscs-planning/issues/13

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by derekbekoe** [comment]
This PR is ready for review. Thanks.
cc: @fengzhou-msft, @haroldrandom

**Comment by yonzhan** [comment]
Codespaces

**Comment by derekbekoe** [comment]
@yonzhan, @fengzhou-msft, @haroldrandom feel free to merge if there are no other comments.
It would be great to have this available Tuesday US time.

**Comment by derekbekoe** [comment]
Thanks @haroldrandom. Doesn't look like I have permission to merge?

**Comment by derekbekoe** [comment]
Is there anything else the PR is being held for? Thanks.

**Comment by yonzhan** [comment]
@fengzhou-msft please help review.

File: src/codespaces/README.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Microsoft Azure CLI 'codespaces' Extension
+==========================================
+
+This package is for the 'codespaces' extension.
+i.e. 'az codespaces'
```
Comment (fengzhou-msft) [inline src/codespaces/README.rst:5]: Can you add more introduction to this extension or a link to the service documentation?

Comment (derekbekoe) [reply src/codespaces/README.rst]: Added link to information on the service.

**Comment by derekbekoe** [comment]
Can we merge now?

File: src/codespaces/azext_codespaces/commands.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+def load_command_table(self, _):
+
+    plan_operations = CliCommandType(
+        operations_tmpl='azext_codespaces.vendored_sdks.vsonline.operations.plan_operations#PlanOperations.{}',
+        client_factory=cf_codespaces)
+
+    with self.command_group('codespace plan', plan_operations, client_factory=cf_codespaces_plan) as g:
+        g.custom_command('list', 'list_plans')
+        g.custom_command('create', 'create_plan')
+        g.command('show', 'get')
```
Comment (fengzhou-msft) [inline src/codespaces/azext_codespaces/commands.py:25]:
```suggestion
        g.show_command('show', 'get')
```

File: src/codespaces/azext_codespaces/commands.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        g.command('delete', 'delete')
+
+    with self.command_group('codespace', plan_operations, client_factory=cf_codespaces_plan) as g:
+        g.custom_command('list', 'list_codespaces', table_transformer=transform_codespace_list_output)
+        g.custom_command('show', 'get_codespace', table_transformer=transform_codespace_item_output)
```
Comment (fengzhou-msft) [inline src/codespaces/azext_codespaces/commands.py:30]:
```suggestion
        g.custom_show_command('show', 'get_codespace', table_transformer=transform_codespace_item_output)
```

File: src/codespaces/azext_codespaces/commands.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        g.custom_command('create', 'create_codespace', table_transformer=transform_codespace_item_output)
+        g.custom_command('open', 'open_codespace')
+        g.custom_command('delete', 'delete_codespace', confirmation="Are you sure you want to delete this Codespace?")
+        g.custom_command('resume', 'resume_codespace', table_transformer=transform_codespace_item_output)
+        g.custom_command('suspend', 'suspend_codespace', table_transformer=transform_codespace_item_output)
+
+    with self.command_group('codespace location', plan_operations, client_factory=cf_codespaces_plan) as g:
+        g.custom_command('list', 'list_available_locations', table_transformer=transform_location_list_output)
+        g.custom_command('show', 'get_location_details', table_transformer=transform_location_detail_output)
```
Comment (fengzhou-msft) [inline src/codespaces/azext_codespaces/commands.py:39]:
```suggestion
        g.custom_show_command('show', 'get_location_details', table_transformer=transform_location_detail_output)
```

**Pull request merged**